src/instruments/yokogawa/yokogawa6370.py

```diff
@@ -60,19 +57,36 @@ def __init__(self, parent, idx):
 
         # METHODS #
 
-        def data(self, bin_format=True):
-            cmd = f":TRAC:Y? {self._name}"
+        def _data(self, axis, limits=None, bin_format=True):
```

Comment:
In general, this makes sense to take it out. Intuitively I would have assumed that `self.data` will return both axes. However, still should be compatible backwards, so this solution will work.
One alternative (for discusison) might be to have `data` set `axis="Y"` by default, however, that it can be switched to `axis="X"` or `axis="both"`. Not sure about this though ;)

In a general use scenario: Do you want to generally just the "Y" data without the wavelength? I haven't used this instrument before, but could imagine that I want both axes in my program to be returned. Ideas?

Reply:
It requires some time to transfer both axes. As you might do the same scan over and over again (to monitor the spectrum), you might want to just copy the "Y" data while the "X" axis (wavelength) remains the same.

Reply:
Thanks for the explanation. Makes a lot more sense now 😃

Reply:
I leave it as is.